gen/field.go

```diff
@@ -100,12 +103,36 @@ func (f fieldGroupGenerator) DefineStruct(g Generator) error {
 			// optional collections because omitempty doesn't differentiate
 			// between nil and empty collections.
 
+			jsonTags := ""
+			jsonFieldName := f.Name
+
 			if (isStructType(f.Type) || isPrimitiveType(f.Type)) && !f.Required {
-				return fmt.Sprintf("`json:\"%s,omitempty\"`", f.Name)
+				jsonTags += ",omitempty"
+			}
+
+			if f.Required {
+				jsonTags += ",required"
+			}
+
+			goTag := f.Annotations["go.tag"]
+			if goTag != "" {
+				structTag := reflect.StructTag(goTag)
+				jsonTag := structTag.Get("json")
+				if jsonTag != "" {
```

Comment:
For the JSON options, let's build a structure from the information in the JSON tag when we're doing all this work.

```
type jsonTag struct {
   FieldName string
  Options map[string]struct{}
}
```